src/requirements/Captcha/hooks/useVerifyCaptcha.ts

```diff
@@ -15,7 +15,7 @@ const useVerifyCaptcha = () => {
   const showErrorToast = useShowErrorToast()
   const toast = useToast()
 
-  const { mutate } = useAccess()
+  const { onSubmit: onJoin } = useJoin()
```

Comment:
I think we should rename useJoin to useMembershipUpdate, and onJoins like this to triggerMembershipUpdate, wdyt?

Reply:
You mean to rename the import, and the returned value like:
```ts
import useMembershipUpdate from "...";
const { onSubmit: triggerMembershipUpdate } = useMembershipUpdate()
```

Or to crate a separate hook named `useMembershipUpdate`, and return onSubmit as `triggerMembershipUpdate`?

Reply:
Sry I mean I'd rename the whole useJoin.tsx file and the exported hook to useMembershipUpdate, since it's not really just join anymore, just wrote the comment here cause it illustrates the motivation well

Reply:
Now that I looked at useJoin there's still a reasonable amount of join specific logic, so ideally we'd separate it into two files, an useMembershipUpdate with the core logic that we'd use everywhere, and an useJoin above that with all the join specific logic that we'd use in JoinModal, and later in other places where we have custom join logic now too: https://discord.com/channels/697041998728659035/1202271479744438303/1202385210264522823